Git Basics: overhaul wording about git commit -m (#25210)
TheOdinProject/curriculum: git/foundations_git/git_basics.md

### Changing the Git Commit Message Editor

If you are using _Visual Studio Code_ (and you should be if you're following this curriculum), there's a way to ensure that if you use `git commit` without the message flag (`-m`), you won't get stuck writing your commit message in in [Vim](<https://en.wikipedia.org/wiki/Vim_(text_editor)>).

> Setting this up is a good idea in case you accidentally omit the flag, but also can just be generally useful.  There is no downside to doing so (unless you love using Vim as your text editor.  But most learners prefer using VS Code).

Reviewer: The last sentence seems a bit out of place, but I couldn't quite put my finger on it. Here is a suggestion, that you can use, or figure out a way for the "But... " to fit better and not in parentheses.
```suggestion
Setting this up is a good idea in case you accidentally omit the flag, but also can just be generally useful. Unless you love using Vim, there is no downside to changing your git commit message editor.  A lot of learners prefer writing commit messages in the comfort of VS Code.
```

> Setting this up leaves you the option to write your commit messages via either the terminal or VS Code.

Reviewer: If you want this to be it's own paragraph, I believe you need to add a new line between the paragraphs.

Reviewer: ```suggestion

Setting this up leaves you the option to write your commit messages via either the terminal or VS Code.
```

> This command should set the configuration.  Type (or copy & paste) this command into your terminal and hit Enter (do include the quotes): 

Reviewer: As a nit pick, I do not think the word "Enter" needs to be capitalized. In addition, I do not think this needs the hint in parentheses after moving the command to a code block. 

Feel to to make these changes as you think makes the most sense to you. I think these two sentences could be combined.
```suggestion
Type (or copy & paste) the following command to set the configuration:
```

> `git config --global core.editor "code --wait"`.

Reviewer: nit pick (non-blocking): I am not sure why "Enter" is capitalized here. I think saying "Enter (or copy & paste) this command into your terminal" might be more succinct.

nit pick: I think "(do include the quotes)" could be removed, after moving the command into a code block. However if you feel strongly about keeping it, I would move it out of the sentence, since this sentence has 2 sets of parentheses.

There will be no confirmation or any output on the terminal after entering this command. 

With that done, you can now choose to use either `git commit -m <your message here>` or `git commit` and enter your message with Visual Studio Code!

> To make a commit with Visual Studio Code as the text editor, just type `git commit` and nothing else after that. Once you do this, a new tab will open. Now you can write your message, and optionally provide more information if you want, right below it. After typing your commit message, save it (Ctrl + S or Mac equivalent) and exit the tab.

Reviewer: I like the mention about being able to provide more information, but I could see people just writing a really long commit message. I feel like we need to be more specific. I also added what they should expect to see in terminal afterwards.
```suggestion
To make a commit with Visual Studio Code as the text editor, just enter `git commit` and nothing else after that. After you hit enter, a new tab in VS Code will open for you to write your commit message. You may provide more details on multiple lines as part of your commit message. After typing your commit message, save it (Ctrl + S or Mac equivalent) and close the tab. If you return to the command line, you will see your commit message and a summary of your changes.
```

### Changing the Git Commit Message Editor

> If you are using _Visual Studio Code_ (and you should be if you're following this curriculum), there's a way to ensure that if you use `git commit` without the message flag (`-m`), you won't get stuck writing your commit message in in [Vim](<https://en.wikipedia.org/wiki/Vim_(text_editor)>).

Reviewer: ```suggestion
If you are using _Visual Studio Code_ (and you should be if you're following this curriculum), there's a way to ensure that if you use `git commit` without the message flag (`-m`), you won't get stuck writing your commit message in [Vim](<https://en.wikipedia.org/wiki/Vim_(text_editor)>).
```
